src/test/java/com/training/api/tests/cat/CatApiTests.java

```diff
@@ -1,79 +1,95 @@
 import java.util.Map;
 
 public class CatApiTests {
     private Logger logger = LoggerFactory.getLogger(this.getClass());
     protected Map<String, String> map = new JsonReaderUtility().getMap();
-    RequestSpecification httpRequest;
+    Map<String, Object> headerMap;
 
     @BeforeEach
     public void setup() {
         RestAssured.baseURI = map.get("baseUri");
-        httpRequest = RestAssured.given();
-        httpRequest.header(map.get("catAuthenticationHeaderName"), map.get("catAuthenticationHeaderValue"));
+        headerMap = new HashMap();
+        headerMap.put(map.get("catAuthenticationHeaderName"), map.get("catAuthenticationHeaderValue"));
     }
 
     @ParameterizedTest
     @CsvFileSource(resources = "/getIdResponse.csv")
-    public void testGetIdResponse(String name, String breedId) {
-
-        Response response = httpRequest.get("/v1/breeds");
-
-        Assertions.assertEquals(response.getStatusCode(), 200);
-
-        List<String> breedName = response.jsonPath().getList("name");
-        int index = breedName.indexOf(name);
-        String id = response.jsonPath().getString(String.format("id[%s]",index));
-        logger.info(id);
-        Assertions.assertEquals(breedId, id,"Id incorrect");
+    public void testGetBreedId(String name, String breedId) {
+        String id = "";
+        io.restassured.response.Response response = new RestEngine().getResponse("/v1/breeds",headerMap);
+        Assertions.assertEquals(200, response.getStatusCode(), "Request Unsuccessful");
+
+        List<com.training.pojos.cat.breeds.Response> breedName = Arrays.asList(response.as(com.training.pojos.cat.breeds.Response[].class));
+        for (int index = 0; index < breedName.size(); index++) {
+            if (breedName.get(index).getName().equalsIgnoreCase(name)) {
+                id = breedName.get(index).getId();
+                logger.info(id);
+            }
+        }
+
+        Assertions.assertEquals(id, breedId, "Id incorrect");
     }
 
     @ParameterizedTest
     @CsvFileSource(resources = "/getWikipediaUrl.csv")
     public void testGetWikipediaUrl(String id, String url) {
-
         String path = String.format("/v1/images/search?breed_ids=%s", id);
-        Response response = httpRequest.when().get(path);
+        io.restassured.response.Response response = new RestEngine().getResponse(path,headerMap);
         Assertions.assertEquals( 200,response.getStatusCode(),"Request Unsuccessful");
 
-        String wikipediaUrl = response.jsonPath().getString("breeds[0].wikipedia_url");
-        Assertions.assertTrue(wikipediaUrl.contains(url), "Url is not correct");
+        List<com.training.pojos.cat.search.Response> responseList =
+                Arrays.asList(response.as(com.training.pojos.cat.search.Response[].class));
+
+        String wikipediaUrl= responseList.get(0).getBreeds()[0].getWikipedia_url();
+        logger.info(wikipediaUrl);
+        Assertions.assertEquals(url,wikipediaUrl,"Response Not Correct");
```

Comment:
can use responseList.get(0).getBreeds()[0].getWikipedia_url() directly intead creating new variable Wikipediaurl

Reply:
using the string wikipedia_url at 2 places. so declared the variable so that code will be less clumsy